### ISSUE TITLE
./vttcompile -a fails on the vttdemo.ttf on linux

### ISSUE THREAD

**Issue by HinTak** [microsoft/VisualTrueType#20]
Just trying vttcompile on the vttdemo.tff shipped with VTT. -s -b and -c completes with success, but -a fails with:

```
$ ./vttcompile -a ../Sample/vttdemo.ttf /tmp/t.ttf

Microsoft Visual TrueType Command Line Interface Version 6.35 
Copyright (C) Microsoft Corporation. Licensed under the MIT License.
....................
...............
Glyf Pgm, glyph 341 (Unicode 0xb8), line 10: NN

Can not complete compile operation! 
```

This is after #17 and fixing #19 . Given #19 (it was surprising to see vttcompile failing on the shipped ttf with the maxp error, and that turns out to be a genuine bug), I'd assume the vtt shipped vttdemo.ttf should be valid and can be used as input to vttcompile for all operations. If it isn't, it should - that's still a bug, but a bug in the font, so the issue should still be reported.

**Comment by HinTak** [comment]
The text parser and etc on unix is quite buggy - I ran a 32-bit cross-compiled build under wine, and it returns success:

```
Z:\home\Hin-Tak\git-others\VisualTrueType\src>vttcompile.exe -a ../Sample/vttdemo.ttf w2.ttf
Microsoft Visual TrueType Command Line Interface Version 6.35
Copyright (C) Microsoft Corporation. Licensed under the MIT License.
....................
...............
Success

Z:\home\Hin-Tak\git-others\VisualTrueType\src>
```

**Comment by HinTak** [comment]
For some reason, -m32 also works allow the result to run to completion.

And about the missing characters, besides \r vs \n (in the code it is sometimes \x0d vs \x0a), it should be said that wchar in microsoft land is 16-bit and same size as short, but on gcc/linux/glibc, it is 32-bit and also int.

**Comment by HinTak** [comment]
The reason why -m32 works is because some of the code use %li to print to a buffer then parse it back. That's is an implicit cast to "long". Anyway, all fix in #22 .

**Comment by HinTak** [comment]
Actually mingw turns out to use the same 16-bit for wchar as msvc; but  it uses only fwprintf, fprintf, vfprintf, wprintf_s from the microsoft runtime; so all the swprintf() are still internal (and seem to follow other gcc convention, rather than windows convention).

**Comment by HinTak** [comment]
fixed with #22 - https://github.com/microsoft/VisualTrueType/commit/e0d3cfa4ce7b5951f040515771c0604f9b906a8e
